### PR TITLE
Rd 207 vcloud docs

### PR DESCRIPTION
**Pull request by EarthmanT**

### CODE REVIEW

File: content/working_with/official_plugins/Infrastructure/vcloud.md [hunk 1]
```diff
@@ -10,407 +10,706 @@ aliases:
     - /developer/official_plugins/vcloud/
 ---
 
-The vCloud plugin enables you to use a vCloud-based infrastructure for deploying services and applications.
+The vCloud Plugin 2 enables you to use a vCloud-based infrastructure for deploying services and applications.
```
Comment (AdarShaked) [inline content/working_with/official_plugins/Infrastructure/vcloud.md:13]: what do you mean by:`Plugin 2`?

File: content/working_with/official_plugins/Infrastructure/vcloud.md [hunk 2]
```diff
@@ -10,407 +10,706 @@ aliases:
...
+    * client config parameters: ['uri', 'api_version', 'verify_ssl_certs', 'log_file', 'log_requests', 'log_headers', 'log_bodies']
+    * credentials parameters: ['user', 'password', 'org']
+  * `resource_config`: A dictionary with required and common parameters to the resource's create or put call.
+  * `use_external_resource`: Boolean. The default value is `false`. Set to `true` if the resource already exists.
+  * `resource_id`: The name of an existing resource in vCloud. Required if `use_external_resource` is `true`. If not set, this will default to the node instance ID.
 
+## Common Runtime Properties
 
-# Types
+The vCloud Plugin stores runtime properties reflecting important data about the resource in VCD. These are the commond properties:
```
Comment (AdarShaked) [inline content/working_with/official_plugins/Infrastructure/vcloud.md:58]: commond -> common

File: content/working_with/official_plugins/Infrastructure/vcloud.md [hunk 3]
```diff
@@ -10,407 +10,706 @@ aliases:
...
+      `logging_enabled`: Logging enabled.
+      `enabled`: Enabled.
+      `vnic`: Interface of VNIC.
+
+**Runtime Properties**
+
+  * `data`:
+    * `gateway_address`: The gateway IP.
+  * `rules`: A dictionary of rules. Each dict accepts these keys:
+    * `ID`: The rule ID.
```
Comment (AdarShaked) [inline content/working_with/official_plugins/Infrastructure/vcloud.md:122]: Twice `ID`?

File: content/working_with/official_plugins/Infrastructure/vcloud.md [hunk 4]
```diff
@@ -10,407 +10,706 @@ aliases:
...
-* `use_external_resource` A boolean for setting whether to create the resource or use an existing one. Defaults to `false`.
-* `resource_id` - The name to give to the new resource, or the name or ID of an existing resource when the `use_external_resource` property is set to `true`. Defaults to `''` (empty string).
-* `vcloud_config` See the [vCloud Configuration](#vcloud-configuration).
+**Properties**
 
-**Mapped Operations:**
+  * `resource_id`: Required. The name of the gateway in VCD.
+  * `resource_config`: Required. A list of dicts. Each dict accepts these keys:
+    - `ip_range`: dnat or snat. Required.
+      `auto_config_dns`: Auto configuration of DNS Default
```
Comment (AdarShaked) [inline content/working_with/official_plugins/Infrastructure/vcloud.md:170]: Add .

File: content/working_with/official_plugins/Infrastructure/vcloud.md [hunk 5]
```diff
@@ -10,407 +10,706 @@ aliases:
...
+      `default_gateway`: The default gateway ip
```
Comment (AdarShaked) [inline content/working_with/official_plugins/Infrastructure/vcloud.md:171]: Add.

File: content/working_with/official_plugins/Infrastructure/vcloud.md [hunk 6]
```diff
@@ -10,407 +10,706 @@ aliases:
...
+Create and delete static routes in the gateway. As this resource is on the gateway, it also preserves the runtime properties of `cloudify.nodes.vcloud.Gateway`.
 
+**Properties**
 
-## cloudify.vcloud.nodes.FloatingIP
+  * `resource_id`: Required. The name of the gateway in VCD.
+  * `resource_config`: Required. A list of dicts. Each dict accepts these keys:
+    - `network`: dnat or snat. Required.
+      `next_hop`: The next hop in the route.
+      `mtu`: The MTU
```
Comment (AdarShaked) [inline content/working_with/official_plugins/Infrastructure/vcloud.md:218]: Add .

**Review by AdarShaked — CHANGES_REQUESTED** [review]

File: content/working_with/official_plugins/Infrastructure/vcloud.md (hunk 1 above)
Comment (EarthmanT) [reply content/working_with/official_plugins/Infrastructure/vcloud.md:13]: It is v2.X and replaces the Tosca plugin which is v1.X.

**Review by AdarShaked — APPROVED** [review]

File: content/working_with/official_plugins/Infrastructure/vcloud.md [hunk 7]
```diff
@@ -10,407 +10,706 @@ aliases:
...
 
+  * `resource_id`: The name of the existing resource. If the `resource_id` was provided as a node property, then this will be the same value. (Except for virtual machines, for more information, please see cloudify.nodes.vcloud.VM below.) Otherwise, this is the node instance ID of the given node instance.
+  * `data`: Please see the runtime property section of each resource for specific keys.
+  * `tasks`: Stores information about tasks that have been executed on the node instance.
 
-## cloudify.vcloud.nodes.Server
+# Node Types
 
-**Derived From:** cloudify.nodes.Compute
+Each node type refers to a resource in VCloud.
```
Comment (ofercloudify) [inline content/working_with/official_plugins/Infrastructure/vcloud.md:68]:
```suggestion
Each node type refers to a resource in vCloud.
```

File: content/working_with/official_plugins/Infrastructure/vcloud.md [hunk 8]
```diff
@@ -10,407 +10,706 @@ aliases:
...
 ---
 
-The vCloud plugin enables you to use a vCloud-based infrastructure for deploying services and applications.
+The vCloud Plugin 2 enables you to use a vCloud-based infrastructure for deploying services and applications.
 
 
 # Plugin Requirements
 
-* Python version 2.7.x
+* Cloudify 5.1 or greater.
```
Comment (ofercloudify) [inline content/working_with/official_plugins/Infrastructure/vcloud.md:18]:
```suggestion
* {{< param cfy_manager_name >}} 5.1 or greater.
```

File: content/working_with/official_plugins/Infrastructure/vcloud.md [hunk 9]
```diff
@@ -10,407 +10,706 @@ aliases:
...
-        * `script_executor` - The script executor. The default is `/bin/bash`.
-    * `hardware` - The key-value hardware customization section, including:
-        * `cpu` - The VM CPU count.
-        * `memory` - The VM memory size, in MB.
-* `management_network` - The management network name.
-* `vcloud_config` See the [vCloud Configuration](#vcloud-configuration).
+## **cloudify.nodes.vcloud.Gateway**
 
-**Mapped Operations:**
+Allows Cloudify to verify that a given gateway exists, to store the gateway's properties in runtime properties, and to allow other resources to interact with it.
```
Comment (ofercloudify) [inline content/working_with/official_plugins/Infrastructure/vcloud.md:73]:
```suggestion
Allows {{< param product_name >}} to verify that a given gateway exists, to store the gateway's properties in runtime properties, and to allow other resources to interact with it.
```

**Review by ofercloudify — CHANGES_REQUESTED** [review]

**Review by ofercloudify — APPROVED** [review]

**Pull request merged**